src/components/LeftSidebar/LeftSidebar.vue

```diff
@@ -203,19 +264,31 @@ export default {
 			preventFindingUnread: false,
 			roomListModifiedBefore: 0,
 			forceFullRoomListRefreshAfterXLoops: 0,
+			isFocused: false,
+			isFiltered: null,
 		}
 	},
 
 	computed: {
 		conversationsList() {
 			let conversations = this.$store.getters.conversationsList
-
-			if (this.searchText !== '') {
-				const lowerSearchText = this.searchText.toLowerCase()
-				conversations = conversations.filter(conversation =>
-					conversation.displayName.toLowerCase().includes(lowerSearchText)
-					|| conversation.name.toLowerCase().includes(lowerSearchText)
-				)
+			switch (this.isFiltered) {
+			case ('unread'):
+				conversations = conversations.filter(conversation => conversation.unreadMessages > 0)
+				break
+			case ('mentions'):
+				conversations = conversations.filter(conversation => conversation.unreadMention || (conversation.unreadMessages > 0
+				 && (conversation.type === CONVERSATION.TYPE.ONE_TO_ONE || conversation.type === CONVERSATION.TYPE.ONE_TO_ONE_FORMER)))
+				break
+			default:
+				if (this.searchText !== '') {
+					const lowerSearchText = this.searchText.toLowerCase()
+					conversations = conversations.filter(conversation =>
+						conversation.displayName.toLowerCase().includes(lowerSearchText)
+								|| conversation.name.toLowerCase().includes(lowerSearchText)
+					)
+					break
+				}
 			}
```

Comment:
Solution for Marco's comment:

> When a filter is active and the user initiates a search, we should just disregard the filter and return the search results normally. Once the user selects a search results we go back to whatever filtering option the user had previously selected.

```suggestion
			if (this.searchText !== '') {
				const lowerSearchText = this.searchText.toLowerCase()
				conversations = conversations.filter(conversation =>
					conversation.displayName.toLowerCase().includes(lowerSearchText)
							|| conversation.name.toLowerCase().includes(lowerSearchText)
				)
			} else if (this.isFiltered === 'unread') {
				conversations = conversations.filter(conversation => conversation.unreadMessages > 0)
			} else if (this.isFiltered === 'mentions') {
				conversations = conversations.filter(conversation => conversation.unreadMention || (conversation.unreadMessages > 0
					&& (conversation.type === CONVERSATION.TYPE.ONE_TO_ONE || conversation.type === CONVERSATION.TYPE.ONE_TO_ONE_FORMER))
			}
```